### DISCUSSION TITLE
Add option to delete respective author when a user is deleted

### DISCUSSION THREAD

**Discussion by andergmartins** [publishpress/PublishPress-Authors#1218] [Feature Requests]
If an author is mapped to a user and that user is deleted it is converted to a guest author. Some clients want to automatically delete the author if its user was deleted. That way they can use bulk actions to delete multiple users at once.

Add an option for changing the default behavior.

**Comment by stevejburge** [comment]
Related issues:
https://github.com/publishpress/PublishPress-Authors/issues/87
https://github.com/publishpress/PublishPress-Authors/issues/67

**Comment by andergmartins** [comment]
We should allow changing the author, or keep them as guest authors (the current behavior).

Checklist:

* [x] Add a filter to `users_have_additional_content` for recognizing the content as secondary authors
* [x] Inject a JS code to replace the user list with an author list (excluding the selected authors)
* [x] Fix edit user form when the user doesn't have posts
* [x] Use a custom delete action to bypass the WP flow so we can change the post authors
* [ ] Refactor the Delete all content to recognize the author ID and not the User ID
* [ ] Add a new option to keep them as guest authors (not removing the author profile). This option keeps the content and delete the users.

**Comment by andergmartins** [comment]
## UI

### No content and no author

```
You have specified these users for deletion:

ID #1: username1
ID #2: username2

[Confirm Deletion]
```

### No content and at least one author

```
You have specified these users for deletion:

ID #1: username1
ID #22: username2

At least one of those users is an author. What should be done with the author terms?

[ ] Delete the author terms
[ ] Convert to guest authors

[Confirm Deletion]
```

### Has content

```
You have specified these users for deletion:

ID #1: username1
ID #22: username2

What should be done with the content owned by this/these users?

[ ] Delete all content where they are first authors
[ ] Delete all content
[ ] Attribute all content to: [      ]
[ ] Keep the content but convert them into guest authors

[Confirm Deletion]
```

**Comment by andergmartins** [comment]
## Test cases

We need to do the tests for single user deletion or bulk user deletion.

### When the selected users have no content and are not authors

  - [x] I see:
    - [x] The user IDs in the list for deletion
    - [x] The Confirm Deletion button
  - [x] I click on Confirm Deletion button and all the users are deleted

### When the selected users have no content but at last one is author

  - [x] I see:
    - [x] The user IDs in the list for deletion
    - [x] The option Delete the author terms
    - [x] The option Convert to guest authors
    - [x] The Confirm Deletion button
  - [x] I select the option to Delete Author Terms then all the users are deleted and the authors terms are deleted
  - [x] I select the option to Convert to guest authors then all the users are deleted and all the author terms are kept as guest authors

### When the selected users have content

  - [x] I see:
    - [x] The user IDs in the list for deletion
    - [x] The option Delete all content when the user is the first author
    - [x] The option Delete all content
    - [x] The option Attribute all content to an author
    - [x] Keep the content but convert the author into a guest author
  - [ ] I select the option Delete all content when the user is the first author then the users are deleted, and the authors are deleted, and the posts where they are first author are deleted, and the other posts are not deleted
  - [ ] I select the option Delete all content and the users are deleted then the author terms are deleted, and all the posts are deleted
  - [ ] I select the option Attribute all content to an author then all the users are deleted, and the authors are deleted, and the posts were attributed to the selected author in the correct order
  - [ ] I select the option Keep the content but convert the author into a guest author then the users are deleted, and the authors are converted to guest authors, and the content is not deleted, and the content has the same authors
